### ISSUE TITLE
VSCode plugin: TypeScript Globals continuously disappear after working in project for a few minutes

### ISSUE THREAD

**Issue by EthanStandel** [DanielXMoore/Civet#466]
Once my repo passed 250 LOC, every 5 minutes or so I have to restart VSC because all of the implicit types/values like Array, window, document, Math, etc just seem to go missing.

![image](https://user-images.githubusercontent.com/16840301/226206375-93503011-e4bf-42fa-8f4d-401a7f0d77e1.png)

**Comment by edemaine** [comment]
Interesting; I haven't seen this on my 600-line codebase.

I'm guessing there's an error message in the log somewhere, at the transition point where everything fails. If you can find it, please post it. Otherwise we'll probably need a reproduction file.

**Comment by EthanStandel** [comment]
@edemaine For the record, the actual compilation here is fine. But I think it's because the TypeScript types associated with the Civet extension appear to be coming from somewhere other than the built in types.

![image](https://user-images.githubusercontent.com/16840301/226207170-f5b8f320-afdb-40ec-abf5-1940d01b69f7.png)

Probably also worth mentioning that I'm currently using Windows + WSL.

**Comment by edemaine** [comment]
I understand that it's just an LSP issue, but I'm not sure I can reproduce it, so it might help to have a 600-line codebase that triggers it, if you can share one.  In addition, I imagine the LSP is producing an error message before this happens, so it'd be great if you can find that in the VSCode logs.

I don't think Civet is doing anything fancy with TS standard lib. It should be coming from the usual place. (But maybe I'm wrong.)

**Comment by EthanStandel** [comment]
@edemaine If you want to try to reproduce it, here's the little project I happen to be trying Civet on

https://github.com/EthanStandel/brick-breaker

I've found a quick way to force the issue is if I open up `src/state/brickState.civet` and then Ctrl/Cmd+Click on a reference to `Math`. Then all global types break in all Civet files in that repo until I restart VSCode. I assume the reason I appear to be running into it randomly over time is because when I start the project, make changes, TypeScript actually goes to validate certain types as I write code, and gets mixed up on the types.

I also just opened up another TypeScript project to validate it doesn't happen there on the same machine. It doesn't, it's only with this Civet repo.

**Comment by edemaine** [comment]
Hmm, perhaps then it's an issue with reference following via Ctrl/Cmd+Click?

I tested this on my machine (Windows 11), and I find that reference following doesn't work: it brings up an empty pop-up, `lib.es2015.core.d.ts` with path `\Users\edemaine\.vscode\extensions\danielx.civet-0.3.6\dist\lib`. I think reference following just doesn't work currently on Windows because of the backslashes. Perhaps because it doesn't work, the LSP doesn't crash. 😅 

Do you see crashing behavior when you *don't* ever use reference following via Ctrl/Cmd+Click?

**Comment by EthanStandel** [comment]
It happens on regular type checking too just over time. I don't have to click into anything. And also, I'm not exactly running on Windows because I'm using WSL and VSC is open in remote-mode via the WSL plugin of the remote development pack. So it's sort of closer to running on Linux than it is to running on Windows directly.

WSL plugin: https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-wsl
Remote dev pack: https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack

**Comment by EthanStandel** [comment]
I think I got it! I didn't look into this issue much between last weekend and this weekend, but I was just reading into the errors and it appears that it was _somewhat_ of a "me" problem (or at least it's solvable). So for this repo, I started with the SPA SolidJS starter app which has never given me any unexpected errors and does work fine with TypeScript. I pretty much just added the Civet plugin to Vite and that was it. And it was fine for a while. I don't know why it broke and wasn't just broken the whole time.

What I did to fix it was just added `"lib": ["esnext", "dom"]` to my `tsconfig.json`'s `compilerOptions`. That has seemed to force everything into agreement between TypeScript & Civet files! I think without that standardization the Civet VSC plugin doesn't have a specific place to go look for global types... even though the code itself compiled fine without that change.

Closing this issue, if you have a similar issue try updating yoru `tsconfig.json`!
